pkg/registry/authorization/selfsubjectrulesreview/rest.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package selfsubjectrulesreview
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	authorizationapi "k8s.io/kubernetes/pkg/apis/authorization"
+)
+
+type REST struct {
+	authorizationRulesGetter authorizer.AuthorizationRulesGetter
+}
+
+func NewREST(authorizationRulesGetter authorizer.AuthorizationRulesGetter) *REST {
+	return &REST{authorizationRulesGetter}
+	//return &REST{ruleResolver: ruleResolver}
+}
+
+func (r *REST) New() runtime.Object {
+	return &authorizationapi.SelfSubjectRulesReview{}
+}
+
+// Create registers a given new ResourceAccessReview instance to r.registry.
+func (r *REST) Create(ctx genericapirequest.Context, obj runtime.Object, includeUninitialized bool) (runtime.Object, error) {
+	// the input object has no valuable input, so don't bother checking it.false
+	user, ok := genericapirequest.UserFrom(ctx)
+	if !ok {
+		return nil, apierrors.NewBadRequest("no user present on request")
+	}
+	namespace, _ := genericapirequest.NamespaceFrom(ctx)
```

Comment:
> This still doesn't look right... I'd expect the types to be cluster-scoped, a

It seems weird to have a cluster scoped resource that is only valid in the context of a namespace.  We can't reasonably evaluate permissions across all namespaces because of fanout.  We could give strictly cluster-scoped powers, but that's of limited utility.  You're just trying for parity with SAR?